halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/openstack/OpenstackAccountValidator.java

```diff
@@ -76,19 +112,27 @@ public void validate(ConfigProblemSetBuilder psBuilder, OpenstackAccount account
     lbaasConfig.setPollTimeout(lbaas.getPollTimeout());
 
     try {
-      List<String> regionsList = Arrays.asList(StringUtils.split(regions, ","));
       OpenstackNamedAccountCredentials openstackCredentials = new OpenstackNamedAccountCredentials.Builder()
           .name(account.getName())
+          .environment(environment)
+          .accountType(accountType)
           .authUrl(authUrl)
           .username(username)
           .password(password)
           .projectName(projectName)
           .domainName(domainName)
-          .regions(regionsList)
+          .regions(regions)
           .insecure(insecure)
+          .heatTemplateLocation(heatTemplateLocation)
+          .consulConfig(consulConfig)
           .lbaasConfig(lbaasConfig)
           .userDataFile(userDataFile)
           .build();
+      if (openstackCredentials == null) {
+        return;
```

Comment:
I agree that this code path can't be hit - what's the intent here? Unless `.build()` can return null?